### PR TITLE
[meta](recover) change dropInfo and RecoverInfo to GSON

### PR DESCRIPTION
**Pull request by morningman**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Follow up #13067 

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [x] Yes (If Yes, please explain WHY), must be with #13067 
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morningman** [comment]
cc @nextdreamblue

File: fe/fe-core/src/main/java/org/apache/doris/journal/JournalEntity.java [hunk 1]
```diff
@@ -196,9 +196,7 @@ public void readFields(DataInput in) throws IOException {
                 break;
             }
             case OperationType.OP_DROP_TABLE: {
-                data = new DropInfo();
-                ((DropInfo) data).readFields(in);
-                isRead = true;
+                data = DropInfo.read(in);
```
Comment (nextdreamblue) [inline fe/fe-core/src/main/java/org/apache/doris/journal/JournalEntity.java:199]:
```suggestion
                data = DropInfo.read(in);
                isRead = true;
```

File: fe/fe-core/src/main/java/org/apache/doris/journal/JournalEntity.java [hunk 2]
```diff
@@ -238,14 +236,11 @@ public void readFields(DataInput in) throws IOException {
             case OperationType.OP_RECOVER_DB:
             case OperationType.OP_RECOVER_TABLE:
             case OperationType.OP_RECOVER_PARTITION: {
-                data = new RecoverInfo();
-                ((RecoverInfo) data).readFields(in);
-                isRead = true;
+                data = RecoverInfo.read(in);
```
Comment (nextdreamblue) [inline fe/fe-core/src/main/java/org/apache/doris/journal/JournalEntity.java:239]:
```suggestion
                data = RecoverInfo.read(in);
                isRead = true;
```

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 38.35 seconds
 load time: 567 seconds
 storage size: 17154655343 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221031155621_clickbench_pr_36577.html

**Review by yiguolei — APPROVED** [review]
lgtm

**Pull request merged**